addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+import * as glob from 'glob';
+import * as path from 'path';
+const Promise = require('ember-cli/lib/ext/promise');
+
+/**
+ * Declare an Interface where key is a string and value is either a string or a number.
+ * Use for storing a string literal, start position and end posittion of ImportClause node kinds.
+ */
+interface ModuleImport {
+  specifier: string;
+  pos: number;
+  end: number;
+};
+
+export interface ModuleMap {
+  [key: string]: ModuleImport;
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+function createTsSourceFile(fileName: string): Promise<ts.SourceFile> {
+  const readFile = Promise.denodeify(fs.readFile);
+  return readFile(fileName, 'utf8').then((contents: string) => {
+    return ts.createSourceFile(fileName, contents, ts.ScriptTarget.ES6, true);
+  });
+}
+
+/**
+ * Traverse through ts.Node recursively and stores moduleSpecifiers in an array
+ * 
+ * @param {ts.Node} node: Typescript Node of whose AST is being traversed
+ * @param {string[]} moduleSpecifiers: array to store moduleSpecifiers if found.
+ * 
+ */
+function _getImportClauses(node: ts.SourceFile): ModuleImport[] {
+  let moduleSpecifiers: ModuleImport[] = [];
+  node.statements.forEach((nodeObj: ts.Node) => {
+    // Every object in the node.statements has an child node object called moduleSpecifier
+    // TODO: investigate on the cases where this assumption is not true.
```

Comment:
This assumption is _definitely_ not true. Only import statements will have module specifiers.